chainindex/indexer.go

```diff
@@ -0,0 +1,450 @@
+	}
+
+	si.hasTipsetStmt, err = si.db.Prepare(stmtHasTipset)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "hasTipsetStmt", err)
+	}
+
+	si.updateTipsetToNonRevertedStmt, err = si.db.Prepare(stmtUpdateTipsetToNonReverted)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "updateTipsetToNonRevertedStmt", err)
+	}
+
+	si.updateTipsetToRevertedStmt, err = si.db.Prepare(stmtUpdateTipsetToReverted)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "updateTipsetToRevertedStmt", err)
+	}
+
+	si.removeTipsetsBeforeHeightStmt, err = si.db.Prepare(stmtRemoveTipsetsBeforeHeight)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "removeTipsetsBeforeHeightStmt", err)
+	}
+
+	si.removeEthHashesOlderThanStmt, err = si.db.Prepare(stmtRemoveEthHashesOlderThan)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "removeEthHashesOlderThanStmt", err)
+	}
+
+	si.updateTipsetsToRevertedFromHeightStmt, err = si.db.Prepare(stmtUpdateTipsetsToRevertedFromHeight)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "updateTipsetsToRevertedFromHeightStmt", err)
+	}
+
+	si.updateEventsToRevertedFromHeightStmt, err = si.db.Prepare(stmtUpdateEventsToRevertedFromHeight)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "updateEventsToRevertedFromHeightStmt", err)
+	}
+
+	si.isTipsetMessageNonEmptyStmt, err = si.db.Prepare(stmtIsTipsetMessageNonEmpty)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "isTipsetMessageNonEmptyStmt", err)
+	}
+
+	si.getMinNonRevertedHeightStmt, err = si.db.Prepare(stmtGetMinNonRevertedHeight)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "getMinNonRevertedHeightStmt", err)
+	}
+
+	si.hasNonRevertedTipsetStmt, err = si.db.Prepare(stmtHasNonRevertedTipset)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "hasNonRevertedTipsetStmt", err)
+	}
+
+	si.updateEventsToNonRevertedStmt, err = si.db.Prepare(stmtUpdateEventsToNonReverted)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "updateEventsToNonRevertedStmt", err)
+	}
+
+	si.updateEventsToRevertedStmt, err = si.db.Prepare(stmtUpdateEventsToReverted)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "updateEventsToRevertedStmt", err)
+	}
+
+	si.getMsgIdForMsgCidAndTipsetStmt, err = si.db.Prepare(stmtGetMsgIdForMsgCidAndTipset)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "getMsgIdForMsgCidAndTipsetStmt", err)
+	}
+
+	si.insertEventStmt, err = si.db.Prepare(stmtInsertEvent)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "insertEventStmt", err)
+	}
+
+	si.insertEventEntryStmt, err = si.db.Prepare(stmtInsertEventEntry)
+	if err != nil {
+		return xerrors.Errorf("prepare %s: %w", "insertEventEntryStmt", err)
+	}
+
+	return nil
+}
+
+func (si *SqliteIndexer) IndexEthTxHash(ctx context.Context, txHash ethtypes.EthHash, msgCid cid.Cid) error {
+	si.closeLk.RLock()
+	if si.closed {
```

Comment:
I think we should create a separate function for this, it is being used in a lot places.